core/arch/arm/kernel/ree_fs_ta.c

```diff
@@ -183,6 +222,8 @@ static TEE_Result ree_fs_ta_open(const TEE_UUID *uuid,
 error_free_payload:
 	thread_rpc_free_payload(mobj);
 error:
+	if (ehdr)
```

Comment:
This check isn't needed.

Reply:
Okay, will remove this one and one similar check below.